src/main/java/com/hazelcast/remotecontroller/CloudManager.java

```diff
@@ -0,0 +1,428 @@
+package com.hazelcast.remotecontroller;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.*;
+import okhttp3.Response;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import net.lingala.zip4j.ZipFile;
+import org.python.jline.internal.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.*;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class CloudManager {
+    private static final OkHttpClient client = new OkHttpClient();
+    private String baseUrl;
+    private final ObjectMapper mapper = new ObjectMapper();
+    private String bearerToken;
+    private static final Logger LOG = LogManager.getLogger(Main.class);
+    private static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+    private Call call;
+    private static final int timeoutForClusterStateWait = 5;
+    private static final int retryTimeInSecond = 10;
+    private static final int clusterTypeId = 5; // Serverless cluster
+    private static final int cloudProviderId = 1; // aws
+    private static final int cloudProviderRegionId = 4; // us-west-2
+    private static final String clusterPlan = "SERVERLESS"; // us-west-2
+
+    public CloudManager() {
+    }
+
+    public void loginToCloudUsingEnvironment() throws CloudException {
+
+        // StringUtil.isNullOrEmpty() can be used, but somehow it throws class not found exception. Needed to be investigated.
+        String baseUrl = System.getenv("BASE_URL");
+        String apiKey = System.getenv("API_KEY");
+        String apiSecret = System.getenv("API_SECRET");
+        String nullEnvVariables = "";
+        if(baseUrl == null)
```

Comment:
Minor: Can we use braces even for single-line if statements? It improves readibility